### DISCUSSION TITLE
How to support both generic and atom second parameter

### DISCUSSION THREAD

**Discussion by JackZhaoooooo** [pmndrs/jotai#2370] [Bug report]
❌The following code reports an error (Expected 1 arguments, but got 2)
```js
interface UserInfo {
  isLogin: boolean
  uname: string
  face: string
  mid: string
}

export const userInfoAtom = atom<UserInfo>(
	() => ({
		isLogin: false,
		uname: 'xxxxx',
		face: '',
		mid: '',
	}),
	async (_get, _set: any) => {}
)
```
✅Delete「<UserInfo>」
```js
interface UserInfo {
  isLogin: boolean
  uname: string
  face: string
  mid: string
}

export const userInfoAtom = atom(
	() => ({
		isLogin: false,
		uname: 'xxxxx',
		face: '',
		mid: '',
	}),
	async (_get, _set: any) => {}
)
```

👉🏻try it [typescriptlang play](https://www.typescriptlang.org/play?ts=4.9.5#code/JYWwDg9gTgLgBAbzgQxhEAaOBxApjGXKLAZX0KjgF84AzKdOAIgCsIZlgmBuAKF+AA7CrWQBjXHACqAZyIBJQbQiJecOMBkAZCAHMhALjgAjCBAA2uZILVwAroOQhcRmTChDdt0RNfvPtiDAACZ+HoJeVPy4AB6QsHBiEIJu9nJQisoAgmggcAC8KLkAPLIKShAAfAAUvACQ1QCUBZVw1Qj1dZo6+oJGouZyGJ0OTi5wAOQx09MTw3V1PuMTc51BoZOrdVSN88gyAJ6CYm0A+rr4WKdyMEbWB835rQhRjUA)

**Comment by dai-shi** [comment]
Is this what you want? https://tsplay.dev/w6e3RW

**Reply by JackZhaoooooo** [reply]
thanks！
